wasp-app-runner/src/index.ts

```diff
@@ -0,0 +1,61 @@
+import { log } from "./logging.js";
+import { getOptions } from "./cli.js";
+import { checkDependencies } from "./dependencies.js";
+import { processManager } from "./process.js";
+import { setupEnvFiles } from "./env.js";
+import { getWaspcDirAbsPath } from "./path.js";
+import { executeWithDb } from "./db/index.js";
+
+const options = getOptions();
+
+async function installWaspCli() {
+  log("install-wasp-cli", "info", "Installing Wasp CLI globally...");
+
+  return processManager.spawnWithLog({
+    name: "install-wasp-cli",
+    cmd: "cabal",
+    args: ["install", "--overwrite-policy=always"],
+    cwd: getWaspcDirAbsPath(),
+  });
+}
+
+async function main() {
```

Comment:
Hmm, this program's flow is a little difficult to follow.

Who parses the command line options?
`main` relies on `options`, which relies on `getOptions`, which relies on `argv`, which is just some local constant. The control flow implicitly extends from functions' bodies onto global constants instead of going top-down.

Here's what I find intutitive.

I'd expect `main` to call `getOptions`. It can already do this since `options` aren't exported (which is good).

After `main` gets all the CLI options, they should no longer travel together - nothing really connects them other than being configurable by the CLI (which is finished at that point). Functions `executeWithDb` and `setupEnvFiles` shouldn't get or know about all the CLI options anyway.